spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/monitor_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "manageiq/providers/hawkular/middleware_manager/event_catcher/monitor"
+
+describe ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Monitor do
+  subject do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    auth                 = AuthToken.new(:name     => "test",
+                                         :auth_key => "valid-token",
+                                         :userid   => "jdoe",
+                                         :password => "password")
+    ems                  = FactoryGirl.create(:ems_hawkular,
+                                              :hostname        => 'localhost',
+                                              :port            => 8080,
+                                              :authentications => [auth],
+                                              :zone            => zone)
+    described_class.new(ems)
+  end
+
+  context "#each_batch" do
+    # VCR.eject_cassette
+    # VCR.turn_off!(ignore_cassettes: true)
+
+    VCR.configure do |c|
+      c.default_cassette_options = {
+        :match_requests_on => [:method, VCR.request_matchers.uri_without_params(:startTime)]
+      }
+    end
+
+    it "# invoke each_batch" do
+      # if generating a cassette the polling window is the previous 1 minute
+      VCR.use_cassette(described_class.name.underscore.to_s) do # , :record => :new_episodes) do
```

Comment:
This comment is useful to me. You will have to get used to me having some comments in the code, I like comments :) Although I do agree several in this initial code were unnecessary, left over from learning, I have removed most that you recommended.